horovod/mxnet/mpi_ops.cc

```diff
@@ -187,18 +185,20 @@ void DoBroadcastCudaOnCPU(
 #endif
 
 extern "C" int horovod_mxnet_allreduce_async(NDArray* input, NDArray* output,
-                                             char* name, bool average) {
-  auto handle = handle_manager.AllocateHandle();
+                                             const char* name, bool average,
+                                             int* handle) {
```

Comment:
why do we use int* handle here? we cannot return the handle to the caller?

If you are not returning, the function return type should be changed to void

Same applies to other functions.

Reply:
This function returns the status through `MX_API_END()` instead of handle. The return status is actually needed at python level when you call [`check_call()`](https://github.com/ctcyang/horovod/pull/19/files/9b9bab1e80987e1f3ec25a1fc5cbed9f55d30e4c..2d64e05c509241f1aff4e070fe69ee428fd849bd#diff-6626ba53afb59b3f75cb017aa6259fd1R119).